client/state/data-layer/wpcom/sites/automated-transfer/initiate/test/index.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import sinon from 'sinon';
+
+/**
+ * Internal dependencies
+ */
+import {
+	initiateTransfer,
+	receiveResponse,
+	receiveError,
+	updateUploadProgress,
+} from '../';
+import {
+	getAutomatedTransferStatus,
+} from 'state/automated-transfer/actions';
+import {
+	pluginUploadError,
+	updatePluginUploadProgress,
+} from 'state/plugins/upload/actions';
+
+const siteId = 1916284;
+
+const ERROR_RESPONSE = {
+	error: 'invalid_input',
+	message: 'Invalid file type.',
+};
+
+describe( 'initiateTransfer', () => {
+	it( 'should dispatch an http request', () => {
+		const dispatch = sinon.spy();
+		initiateTransfer( { dispatch }, { siteId, pluginZip: 'foo' } );
+		expect( dispatch ).to.have.been.calledWithMatch( {
```

Comment:
> Any good reason to test with calledWithMatch instead of the more precise calledWith?

There are a lot of extra fields, some of which are circular refs, so not actually sure `calledWith` would be possible here.

> the test lacks the second action arg that dispatch is called with

`action` is an arg to `http()` here. I'm happy to check that we're hitting the expected endpoint with the right args.